src/client.rs

```diff
@@ -375,11 +375,9 @@ impl<C: Config> Client<C> {
         self.msgs_to_send.push(msg.into());
     }
 
-    pub fn send_plugin_message(&mut self, channel: Ident, data: BoundedArray<u8, 0, 1048576>) {
-        self.plugin_messages_to_send.push(PluginMessageToClient{
-            channel,
-            data,
-        });
+    pub fn send_plugin_message(&mut self, channel: Ident, data: RawBytes) {
```

Comment:
`RawBytes` is part of the `protocol` module. You can accept a `Vec<u8>` instead.